internal/pkg/template/templates/workloads/partials/cf/nlb.yml

```diff
@@ -10,25 +14,90 @@ PublicNetworkLoadBalancer:
+      Fn::ImportValue:
+        !Sub "${AppName}-${EnvName}-VpcId" 
+    Tags:
+      - Key: Name
+        Value: !Sub 'copilot-${AppName}-${EnvName}-${WorkloadName}-nlb'
+EnvironmentSecurityGroupIngressFromNetworkLoadBalancerSecurityGroup:
+  Type: AWS::EC2::SecurityGroupIngress
+  Metadata:
+    'aws:copilot:description': 'Allow ingress from the network load balancer to services in your environment'
+  Properties:
+    GroupId:
+      Fn::ImportValue:
+        !Sub '${AppName}-${EnvName}-EnvironmentSecurityGroup'
+    IpProtocol: -1
+    SourceSecurityGroupId: !GetAtt NetworkLoadBalancerSecurityGroup.GroupId
+{{- else}}
+NLBSecurityGroup:
+  Metadata:
+    'aws:copilot:description': 'A security group for your network load balancer to route traffic to service'
+  Type: AWS::EC2::SecurityGroup
+  Properties:
+    GroupDescription: Allow access from the network load balancer to service
+    SecurityGroupIngress:
+{{range $cidr := .NLB.PublicSubnetCIDRs}}
+  {{- range $listener := $.NLB.Listener}}
+      - CidrIp: {{$cidr}}
+        Description: Ingress to allow access from Network Load Balancer subnet
+        FromPort: {{ $listener.TargetPort }}
+        IpProtocol: {{- if eq $listener.Protocol "TLS" }} TCP {{- else }} {{ $listener.Protocol }} {{- end}}
+        ToPort: {{ $listener.TargetPort }}
+      {{- if $listener.HealthCheck.Port}}{{- if ne $listener.HealthCheck.Port $listener.Port}}
+      - CidrIp: {{$cidr}}
+        Description: Ingress to allow access from Network Load Balancer subnet for health check
+        FromPort: {{ $listener.HealthCheck.Port }}
+        ToPort: {{ $listener.HealthCheck.Port }}
+        IpProtocol: TCP
+      {{- end}}{{- end}}
+  {{- end}}
+{{end}}
+    Tags:
+      - Key: Name
+        Value: !Sub 'copilot-${AppName}-${EnvName}-${WorkloadName}-nlb'
+    VpcId:
+      Fn::ImportValue:
+        !Sub "${AppName}-${EnvName}-VpcId"
+{{- end}}
+
+{{- if .NLB.UDPListenerExists }}
 {{- range $i, $listener := .NLB.Listener }}
 NLBListener{{- if ne $i 0 }}{{$i}}{{end}}:
   Metadata:
     'aws:copilot:description': 'A {{$listener.Protocol}} listener on port `{{$listener.Port}}` that forwards traffic to your tasks'
   Type: AWS::ElasticLoadBalancingV2::Listener
   Properties:
     DefaultActions:
-      - TargetGroupArn: !Ref NLBTargetGroup{{- if ne $i 0 }}{{$i}}{{end}}
+      - TargetGroupArn: !Ref NetworkLoadBalancerTargetGroup{{- if ne $i 0 }}{{$i}}{{end}}
         Type: forward
-    LoadBalancerArn: !Ref PublicNetworkLoadBalancer
+    LoadBalancerArn: !Ref PublicNetworkLoadBalancerV2
     Port: {{ $listener.Port }}
     Protocol: {{ $listener.Protocol }}
   {{- if eq $listener.Protocol "TLS" }}
     Certificates:
       - CertificateArn: !Ref NLBCertValidatorAction
     SslPolicy: {{ if $listener.SSLPolicy }}{{ $listener.SSLPolicy }}{{ else }} ELBSecurityPolicy-TLS13-1-2-2021-06 {{ end }}
   {{- end}}
-
-NLBTargetGroup{{- if ne $i 0 }}{{$i}}{{end}}:
+NetworkLoadBalancerTargetGroup{{- if ne $i 0 }}{{$i}}{{end}}:
```

Comment:
do we have to recreate the target group too? can we not just recreate the lb?

Reply:
The target group depends on the NLB so unfortunately it has to be recreated.

Reply:
i don't understand 🤔 why do we have to rename it, and therefore, force the target group to be recreated? won't the new load balancer's listener just be able to reference the existing target group?

Reply:
nvm, got it - a target group can't be attached to two listeners at once, which would happen during the deployment